Examples/FileSync/master.py

```diff
@@ -0,0 +1,102 @@
+"""
+Copyright 2018 SPARKL Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Simple implementation of FileSync/Master service.
+"""
+import os
+import sparkl_cli as sparkl
+from watchdog.observers import Observer
+from watchdog.events import FileSystemEventHandler
+from base64 import b64encode
+
+
+class CustomHandler(FileSystemEventHandler):
+  """
+  Watchdog handler callbacks marshal arguments and
+  invoke the functions corresponding to the notify
+  operations on the master service.
+  """
+  def __init__(self, service):
+    self.service = service
+
+  def on_created(self, event):
+    binary = bytearray()
+    if not event.is_directory:
+      with open(event.src_path, mode='rb') as file:
+        binary = file.read()
+
+    put(self.service, event.is_directory, event.src_path, binary)
+
+  def on_deleted(self, event):
+      delete(self.service, event.is_directory, event.src_path)
+
+  def on_modified(self, event):
```

Comment:
on_modified and on_created looks nearly identical

Reply:
Almost. But not quite ;)